lib/Zonemaster/Backend/DB/SQLite.pm

```diff
@@ -196,23 +196,23 @@ sub get_test_history {
         $undelegated = "AND undelegated = 0";
     }
 
-    my $quoted_domain = $self->dbh->quote( $p->{frontend_params}->{domain} );
-    $quoted_domain =~ s/'/"/g;
-    my $query = "SELECT
-                    id,
-                    hash_id,
-                    creation_time,
-                    params,
-                    undelegated,
-                    results
-                 FROM
-                    test_results
-                 WHERE
-                    params like '\%\"domain\":$quoted_domain\%'
-                    $undelegated
-                 ORDER BY id DESC LIMIT $p->{limit} OFFSET $p->{offset} ";
-    my $sth1 = $self->dbh->prepare( $query );
-    $sth1->execute;
+    my @results;
+    my $query = q[
+        SELECT
+            id,
+            hash_id,
+            creation_time,
+            undelegated,
+            results
+        FROM test_results
+        WHERE progress = 100 AND domain = ? ] . $undelegated . q[
```

Comment:
The `IS` operator could be used here to avoid concatenation of SQL syntax.

Reply:
What I had in mind was pretty much what you did here but using SQLite specific syntax for the comparison (i.e `undelegated IS ?`). But I like what you did here better.